src/core/ui/components/drop-down/index.tsx

```diff
@@ -1,17 +1,402 @@
+                        className={summaryStyles.settingsValue}
+                        style={{ opacity: 0.6 }}
+                      >
+                        {shortenEthAddress(web3Wallets[0].displayValue)}
+                      </Text>
+                      <div
+                        data-tip="Check on Etherscan"
+                        data-for="etherscan-tooltip"
+                      >
+                        <Icon
+                          style={{
+                            marginLeft: 8,
+                            height: '18px',
+                            width: '18px',
+                          }}
+                          onClick={() => {
+                            window.open(
+                              `https://etherscan.io/address/${web3Wallets[0].displayValue}`,
+                              '_blank'
+                            );
+                          }}
+                        >
+                          {EtherscanLogo}
+                        </Icon>
+                      </div>
+                      <ReactTooltip
+                        place="top"
+                        type="dark"
+                        effect="solid"
+                        id="etherscan-tooltip"
+                      />
+                      <Icon
+                        style={{
+                          cursor: 'pointer',
+                          height: '18px',
+                          width: '18px',
+                        }}
+                        onClick={() => {
+                          navigator.clipboard.writeText(
+                            web3Wallets[0].displayValue
+                          );
+                          setCopyManagedWalletSuccess(true);
+                          setTimeout(
+                            () => setCopyManagedWalletSuccess(false),
+                            2000
+                          );
+                        }}
+                      >
+                        {copyManagedWalletSuccess ? CheckMarkIcon : copyIcon}
+                      </Icon>
+                    </div>
+                  </div>
+                )}
+                {emails.length ? (
+                  <div
+                    className={summaryStyles.actionOnHover}
+                    key={emails[0].id}
+                  >
+                    <div style={{ display: 'flex', alignItems: 'center' }}>
+                      <ConnectionTypeIcon type={emails[0].connectionType} />
+                      <Text
+                        align={Align.Left}
+                        className={summaryStyles.settingsValue}
+                      >
+                        {emails[0].displayValue}
+                      </Text>
+                      <Chip>Verified</Chip>
+                    </div>
+                    <BinIcon onClick={() => removeConnection(emails[0].id)} />
+                  </div>
+                ) : null}
+                {!nonCustodialWallets.length ? (
```

Comment:
Any chance we can break these up so it's not a giant tree? Perhaps having variables or even sub components? It will make it far easier to debug and read